### DISCUSSION TITLE
How to set applicationname in c# open telemetry app for tanzuobservabilityexporter

### DISCUSSION THREAD

**Discussion by syamalaids** [open-telemetry/opentelemetry-collector-contrib#7834] [Q&A]
Hello,
 The "ReadMe" for tanzuobservabilityexporter states the following:
**Application identity tags, which are [required by Tanzu Observability](https://docs.wavefront.com/trace_data_details.html#how-wavefront-uses-application-tags), are added if they are missing.
application is set to "defaultApp".
service is set to "defaultService".*
  
I am using the following method which is setting the service name:
```
   TracerProvider tracerProvider = Sdk.CreateTracerProviderBuilder()
                                    .AddSource(ServiceName)
                                    .SetResourceBuilder(
                                            ResourceBuilder.CreateDefault()
                                                .AddService(serviceName: ServiceName, serviceVersion: ServiceVersion))
                                    .AddHttpClientInstrumentation(SetInstrumentationOptions)
                                    .AddOtlpExporter(SetExporterOptions)
                                    .Build();
```
Have no clue how to set the "Application Name"? Wavefront defines "ApplicationTags" but that will work with their SDK.
How to achieve the same with OpenTelemetry?

**Comment by jpkrohling** [comment]
cc  @oppegard @thepeterstone @keep94

**Reply by oppegard** [reply]
Hi @syamalaids, take a look at our C# example code for setting the application name via OTel Resource attributes: https://github.com/wavefrontHQ/opentelemetry-examples/blob/master/DOTNET-example/Web-App/Program.cs#L8-L10. On line 17, we add those attributes to the `ResourceBuilder`.

**Reply by syamalaids** [reply]
Thank you that worked!
